src/accel/detail/RootOffloadWriter.hh

```diff
@@ -0,0 +1,91 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/detail/RootOffloadWriter.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <array>
+
+#include "corecel/Macros.hh"
+#include "celeritas/ext/RootFileManager.hh"
+#include "celeritas/phys/Primary.hh"
+
+namespace celeritas
+{
+class ParticleParams;
+
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Export primary data to ROOT.
+ *
+ * One TTree entry represents one primary.
+ */
+class RootOffloadWriter
```

Comment:
Likewise since this class doesn't depend on anything in `accel` and we can use it for writing out whatever primaries, put this in celeritas/io as `RootEventWriter`. The only thing that `OffloadWriter` did was to wrap a class for thread safety.